rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java

```diff
@@ -377,14 +377,14 @@ private void processHelloMessage(ChannelHandlerContext ctx, HelloMessage helloMe
         List<Capability> capInCommon = configCapabilities.getSupportedCapabilities(helloMessage);
         channel.initMessageCodes(capInCommon);
         for (Capability capability : capInCommon) {
-            // It seems that the only supported capability is RSK, and everything else is ignored.
-            if (Capability.RSK.equals(capability.getName())) {
+            // Accept RSK or SNAP capabilities.
+            if (Capability.RSK.equals(capability.getName()) || Capability.SNAP.equals(capability.getName())) {
```

Comment:
It is correct to have an OR condition here? I mean, does it has sense to do snap sync if the node does not support RSK?

Reply:
As discussed in the document [here](https://www.notion.so/iovlabs/Analyse-possible-options-for-SNAP-feature-discovery-among-connected-peers-148f805e06fc4b778cb5bf8a21059a7c), both capabilities are independent from one another. A peer supporting the SNAP protocol does not necessarily need to support the RSK protocol. Our current design doesn't support nested capabilities